aspnetcore/blazor/blazor-server-ef-core.md

```diff
@@ -0,0 +1,126 @@
+
+The factory is injected into components and used to create new instances. For example, in `Pages/Index.razor`:
+
+[!code-csharp[](./common/samples/3.x/BlazorServerEFCoreSample/BlazorServerDbContextExample/Pages/Index.razor?range=199-212)]
+
+### Scope to the component lifetime
+
+There are two approaches to scope a context to the component lifetime. If the context is registered with DI, you can use the special `OwningComponentBase` base class to create a control scope. Learn more about the class [here](https://docs.microsoft.com/dotnet/api/microsoft.aspnetcore.components.owningcomponentbase-1).
+
+The second approach is to use the factory to create a context and track it for the lifetime of the component. First, implement `IDisposable` and inject the factory as shown in `Pages/EditContact.razor`.
+
+[!code-razor[](./common/samples/3.x/BlazorServerEFCoreSample/BlazorServerDbContextExample/Pages/EditContact.razor?range=5-7)]
+
+The sample app ensures the contact is disposed when the component is:
+
+[!code-csharp[](./common/samples/3.x/BlazorServerEFCoreSample/BlazorServerDbContextExample/Pages/EditContact.razor?range=181-184)]
+
+Finally, `OnInitializedAsync` is overridden to create a new context. In the sample app, it loads the contact in the same method.
+
+[!code-csharp[](./common/samples/3.x/BlazorServerEFCoreSample/BlazorServerDbContextExample/Pages/EditContact.razor?range=89-104)]
+
+## The sample app
+
+The sample application was built as a reference for Blazor Server apps the use EF Core. It includes a grid with sorting and filtering, delete, add, and update operations. It demonstrates use of EF Core to handle optimistic concurrency.
+
+[View or download sample code](https://github.com/dotnet/AspNetCore.Docs/tree/master/aspnetcore/blazor/common/samples/3.x/BlazorServerEFCoreSample) ([how to download](xref:index#how-to-download-a-sample))
+
+By default, the sample uses [SQL Server Express LocalDB](https://docs.microsoft.com/sql/database-engine/configure-windows/sql-server-express-localdb). You can override the connection string by updating the connection string in `appsettings.json`:
+
+[!code-json[](./common/samples/3.x/BlazorServerEFCoreSample/BlazorServerDbContextExample/appsettings.json?highlight=3)] 
+
+The sample also configures database logging to show the SQL queries that are generated. This is configured in `appsettings.Development.json`. 
+
+[!code-json[](./common/samples/3.x/BlazorServerEFCoreSample/BlazorServerDbContextExample/appsettings.Development.json?highlight=8)] 
+
+The `Startup.cs` class uses the factory extension to register the context and set the options. 
+
+[!code-csharp[](./common/samples/3.x/BlazorServerEFCoreSample/BlazorServerDbContextExample/Startup.cs?highlight=28-32)]
+
+To make it easier to populate the sample database, a call to ensure it is created and populate random sample contacts is made in `Program.cs`. 
```

Comment:
I would remove this last section about database initialization sample code. You already have a nice comment in the sample pointing out that the database initialization code is just for demonstration purposes. I think calling this out in the doc actually draws more attention to it and creates more opportunity for folks to be confused by it.

Reply:
Note tho that code comments aren't *localized*. We've been moving virtually all code comments to in-text remarks for that reason.

Reply:
I left it out for now.